### PR TITLE
[specific ci=Group3-Docker-Compose]support link in user defined network as well

### LINKED ISSUE #4038: Docker-compose link not working
**VIC version:**

build 8158

**Deployment details:**

What was the vic-machine create command used to deploy the VCH?
```bin/vic-machine-linux create --name=vch-1 --volume-store=datastore1/test:default  --target=root:pwd@192.168.60.162 --thumbprint=...  --debug 1 --no-tlsverify --no-tls```

**Steps to reproduce:**
Here is the docker-compose.yml
```
version: '3'
services:
  db:
    image: busybox
    links:
      - web:aaa
    command: ["ping", "aaa"]
    depends_on:
      - web

  web:
    image: busybox
    container_name: a.b
    command: ["/bin/top"]
```

**Actual behavior:**
The alias does not work. Here is the output from docker-compose logs
```
Attaching to vic_db_1, a.b
db_1   | ping: bad address 'aaa'
```

**Expected behavior:**
"ping aaa" from db should work.

**Additional details as necessary:**
However, if we specify using the default bridge network as the the default network to use in the compose file, it works fine.
```
version: '3'
services:
  db:
    image: busybox
    links:
      - web:aaa
    command: ["ping", "aaa"]
    depends_on:
      - web

  web:
    image: busybox
    container_name: a.b
    command: ["/bin/top"]

networks:
  default:
    external:
      name: bridge
```

### PR DESCRIPTION
**Pull request by emlin**
fix #4038

### CODE REVIEW

File: tests/test-cases/Group3-Docker-Compose/3-03-Docker-Compose-Basic.robot [hunk 1]
```diff
@@ -20,6 +20,7 @@ Suite Teardown  Cleanup VIC Appliance On Test Server
 
 *** Variables ***
 ${yml}  version: "2"\nservices:\n${SPACE}web:\n${SPACE}${SPACE}image: python:2.7\n${SPACE}${SPACE}ports:\n${SPACE}${SPACE}- "5000:5000"\n${SPACE}${SPACE}depends_on:\n${SPACE}${SPACE}- redis\n${SPACE}redis:\n${SPACE}${SPACE}image: redis\n${SPACE}${SPACE}ports:\n${SPACE}${SPACE}- "5001:5001"
+${link-yml}  version: "2"\nservices:\n${SPACE}redis:\n${SPACE}${SPACE}image: redis:alpine\n${SPACE}${SPACE}container_name: redis\n${SPACE}${SPACE}ports: ["6379"]\n${SPACE}web:\n${SPACE}${SPACE}image: busybox\n${SPACE}${SPACE}container_name: a.b.c\n${SPACE}${SPACE}links:\n${SPACE}${SPACE}- redis:aaa\n${SPACE}${SPACE}command: ["ping", "aaa"]
```
Comment (sflxn) [inline tests/test-cases/Group3-Docker-Compose/3-03-Docker-Compose-Basic.robot]: If you use the same name for redis as the first yml, you should call docker-compose down on the first compose app before you bring this one up.  Until we have docker rename, you can't bring up multiple apps that have the same name for services.

**Comment by sflxn** [comment]
lgtm

File: lib/apiservers/engine/backends/container_proxy.go [hunk 2]
```diff
@@ -907,8 +907,8 @@ func toModelsNetworkConfig(cc types.ContainerCreateConfig) *models.NetworkConfig
 		NetworkName: cc.HostConfig.NetworkMode.NetworkName(),
 	}
 
-	// Docker copies Links to NetworkConfig only if it is a UserDefined network, handle that
-	if !cc.HostConfig.NetworkMode.IsUserDefined() && len(cc.HostConfig.Links) > 0 {
+	// Docker support link for bridge network and user defined network, we should handle that
```
Comment (anchal-agrawal) [inline lib/apiservers/engine/backends/container_proxy.go]: s/support/supports

**Comment by anchal-agrawal** [comment]
Merging on behalf of Emma.

**Pull request merged**